.github/workflows/array-api-tests.yml

```diff
@@ -40,7 +40,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ['3.9', '3.10', '3.11', '3.12']
+        python-version: ['3.10', '3.11', '3.12']
```

Comment:
I believe array-api-tests is using a typing feature (|) from Python 3.10, which is causing us to fail here.

Not sure if it's worth patching there, as NEP29 says 3.9 should be dropped anyways.

Reply:
That should be fixed upstream now. Let's keep 3.9 support for now. We should discuss dropping Python support in another issue. I'd prefer to be as conservative as possible with supported Python versions (i.e., not drop support unless we really need to), to maximize compatibility for people.